assets/js/script.js

```diff
@@ -4,126 +4,175 @@
+      incrementCounter("score");
+    } else {
+      alert(
+        `Awwww.... you answered ${userAnswer}. The correct answer was ${
+          this.#correctAnswer
+        }!`
+      );
+      incrementCounter("incorrect");
     }
 
-    // Generate a new question and update the DOM
-    newQuestion() {
-        document.getElementById("answer-box").value = "";
-        document.getElementById("answer-box").focus();
-    
-        [this.#operand1, this.#operand2, this.#correctAnswer] = OPERATIONS[this.#operation].createQuestion();
-
-        document.getElementById('operand1').textContent = this.#operand1;
-        document.getElementById('operand2').textContent = this.#operand2;
-        document.getElementById('operator').textContent = OPERATIONS[this.#operation].symbol;
-    }
-
-    // Check whether the user's answer is correct, update the counters and proceed to the next question
-    checkAnswer() {
-        let userAnswer = parseInt(document.getElementById("answer-box").value);
-
-        if (userAnswer === this.#correctAnswer) {
-            alert("Hey! You got it right! :D");
-            incrementCounter("score");
-        } else {
-            alert(`Awwww.... you answered ${userAnswer}. The correct answer was ${this.#correctAnswer}!`);
-            incrementCounter("incorrect");
-        }
-
-        this.newQuestion()
-    }
+    this.newQuestion();
+  }
 }
 
 // Wait for the DOM to finish loading before running the game
 // Get the button elements and add event listeners to them
-document.addEventListener("DOMContentLoaded", function() {
-    let game = new GameEngine();
-
-    for (let button of document.getElementsByTagName("button")) {
-        if (button.dataset.operation === "submit") {
-            button.addEventListener("click", function() {
-                game.checkAnswer();
-            });
-        } else {
-            button.addEventListener("click", function() {
-                game.operation = this.dataset.operation;
-            });
-        }
+document.addEventListener("DOMContentLoaded", function () {
+  let game = new GameEngine();
+
+  for (let button of document.getElementsByTagName("button")) {
+    if (button.dataset.operation === "submit") {
+      button.addEventListener("click", function () {
+        game.checkAnswer();
+      });
+    } else {
+      button.addEventListener("click", function () {
+        game.operation = this.dataset.operation;
+      });
     }
-
-    document.getElementById("answer-box").addEventListener("keydown", function(event) {
-        if (event.key === "Enter") {
-            game.checkAnswer();
-        }
+  }
+
+  document
+    .getElementById("answer-box")
+    .addEventListener("keydown", function (event) {
+      if (event.key === "Enter") {
+        game.checkAnswer();
+      }
     });
 
-    game.newQuestion();
-});
\ No newline at end of file
+  // Basic asserts
+  console.assert(add(3, 5) === 8, "addition test1 failed");
```

Comment:
I would suggest showing the full details in each assertion, so you can immediately see what's run before having to look at the code. To make this slightly 'somewhat' convenient, `console.assert` supports printf-like substitution. So you can go with something like this:
`console.assert(add(1003, -5) === 998, "Expected %s(%d, %d) to be %d, but got %d", "add", 1003, -5, 998, add(1003, -5));
` 
(and we could use a data structure to provide the inputs dynamically, but I'm not sure if it might be too complicated for students)